rust/src/detect/transform_base64.rs

```diff
@@ -0,0 +1,420 @@
+/* Copyright (C) 2024 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+// Author: Jeff Lucovsky <jlucovsky@oisf.net>
+
+use crate::detect::error::RuleParseError;
+use crate::detect::parser::{parse_var, take_until_whitespace, ResultValue};
+use std::ffi::{CStr, CString};
+use std::os::raw::c_char;
+
+use nom7::bytes::complete::tag;
+use nom7::character::complete::multispace0;
+use nom7::sequence::preceded;
+use nom7::{Err, IResult};
+use std::str;
+
+#[repr(u8)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum DetectBase64Mode {
+    ModeRFC4648,
+    ModeRFC2045,
+    ModeStrict,
+}
+
+pub const TRANSFORM_FROM_BASE64_MODE_DEFAULT: DetectBase64Mode = DetectBase64Mode::ModeRFC4648;
```

Comment:
Can we not implement the `Default` trait because of MSRV ?

Reply:
I tried a few things -- all with failures. Here's one example that failed
```
// Define a trait with a function for the default value
trait DefaultDetectBase64Mode {
    fn default_mode() -> DetectBase64Mode;
}

// Implement the trait for DetectBase64Mode
impl DefaultDetectBase64Mode for DetectBase64Mode {
    fn default_mode() -> DetectBase64Mode {
        DetectBase64Mode::ModeRFC4648
    }
}

// Define your constant using the trait function
pub const TRANSFORM_FROM_BASE64_MODE_DEFAULT: DetectBase64Mode = DetectBase64Mode::default_mode();
```

Reply:
Did you try the `default` attribute cf https://rust-lang.github.io/rfcs/3107-derive-default-enum.html ?

```
#[repr(u8)]
#[derive(Copy, Clone, Debug, Default, PartialEq, Eq)]
pub enum DetectBase64Mode {
    #[default]
    ModeRFC4648,
    ModeRFC2045,
    ModeStrict,
}
```